specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json

```diff
@@ -0,0 +1,114 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Communication",
+    "description": "Communication Services API common types.",
+    "version": "2020-11-19-preview1"
+  },
+  "paths": {},
+  "definitions": {
+    "CommunicationError": {
+      "description": "The Communication Services error exception.",
+      "required": [
+        "error"
+      ],
+      "properties": {
+        "error": {
+          "description": "The Communication Services error.",
+          "$ref": "#/definitions/Error"
+        }
+      }
+    },
+    "Error": {
+      "description": "The Communication Services error.",
+      "required": [
+        "code",
+        "message"
+      ],
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The error code."
+        },
+        "message": {
+          "type": "string",
+          "description": "The error message."
+        },
+        "target": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error target."
+        },
+        "details": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error details."
+        },
+        "innererror": {
+          "x-ms-client-name": "innerError",
+          "readOnly": true,
+          "$ref": "#/definitions/Error",
+          "description": "The inner error if any."
+        }
+      },
+      "x-nullable": true
+    },
+    "CommunicationIdentifierKind": {
+      "description": "The kind of an communication identifier.",
+      "type": "string",
+      "enum": [
+        "unknown",
+        "communicationUser",
+        "phoneNumber",
+        "callingApplication"
+      ],
+      "x-ms-enum": {
+        "name": "CommunicationIdentifierType",
+        "modelAsString": true
+      }
+    },
+    "CommunicationIdentifier": {
+      "description": "Identifies a participant in Azure Communication services. A participant is, for example, a phone number or an Azure communication user.",
+      "required": [
+        "kind",
+        "id"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/CommunicationIdentifierKind",
+          "description": "Kind of the communication identifier."
+        },
+        "communicationUserId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of the CommunicationUser as returned from the Communication Service."
+        },
+        "phoneNumber": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The phone number in E.164 format."
+        },
+        "callingApplicationId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of the CallingApplication."
+        },
+        "id": {
```

Comment:
not sure how `id` and the other id properties would be used.
If kind = `communicationUser` does that mean both `id` and `communicationUserId` would be populated?
Would there be difference in the values, their formatting or encoding?

Reply:
Great question. `id` will always get populated with the actual MRI, but in the client SDK it will get stripped.
However, if the client doesn't understand the `kind` because it's a newly introduced enum value, the client will create an `UnknownIdentifier` and only use the `id` property.

So a phone number for example will set the `phoneNumber` property and also set the `id` with the 4: representation of the phone number.

Reply:
Thanks for the clarification.